src/main/java/eu/tsystems/mms/tic/testerra/plugins/selenoid/webdriver/VideoDesktopWebDriverFactory.java

```diff
@@ -51,32 +59,33 @@
     private final VideoRequestStorage videoRequestStorage = VideoRequestStorage.get();
 
     @Override
-    protected DesktopWebDriverRequest buildRequest(AbstractWebDriverRequest request) {
-        return super.buildRequest(request);
+    public void accept(WebDriver webDriver) {
+        IWebDriverManager webDriverManager = Testerra.getInjector().getInstance(IWebDriverManager.class);
+        webDriverManager.getSessionContext(webDriver).ifPresent(sessionContext -> {
+            // create a VideoRequest with request and videoName
+            final VideoRequest videoRequest = new VideoRequest(sessionContext, sessionContext.getWebDriverRequest().getCapabilities().get(SelenoidCapabilityProvider.Caps.videoName.toString()).toString());
+
+            // store it.
+            videoRequestStorage.store(videoRequest);
+            log().info("VNC Streaming URL: " + selenoidHelper.getRemoteVncUrl(videoRequest));
+        });
     }
 
     @Override
-    public WebDriver getRawWebDriver(DesktopWebDriverRequest request, DesiredCapabilities desiredCapabilities, SessionContext sessionContext) {
+    public void accept(WebDriverRequest webDriverRequest, SessionContext sessionContext) {
+
+        // Only accept webdrivers for desktop
+        if (!(webDriverRequest instanceof DesktopWebDriverRequest)) {
+            return;
+        }
 
         if (VNC_ACTIVE && StringUtils.isBlank(VNC_ADDRESS)) {
             log().warn(String.format("%s is set to true, but vnc host property %s was not set.", SelenoidProperties.VNC_ENABLED, SelenoidProperties.VNC_ADDRESS));
         }
 
+        DesktopWebDriverRequest desktopWebDriverRequest = (DesktopWebDriverRequest)webDriverRequest;
         // determine everything for selenoid... incl. video name on remote.
-        final Capabilities videoCaps = selenoidCapabilityProvider.provide(request);
-        desiredCapabilities = desiredCapabilities.merge(videoCaps);
-
-        // start webdriver with selenoid caps.
-        final WebDriver rawWebDriver = super.getRawWebDriver(request, desiredCapabilities, sessionContext);
-
-        // create a VideoRequest with request and videoName
-        final VideoRequest videoRequest = new VideoRequest(sessionContext, videoCaps.asMap().get(SelenoidCapabilityProvider.Caps.videoName.toString()).toString());
-
-        // store it.
-        videoRequestStorage.store(videoRequest);
-
-        // get vnc path and log
-        log().info("VNC Streaming URL: " + selenoidHelper.getRemoteVncUrl(videoRequest));
```

Comment:
This is a feature to get VNC url if you are running the tests locally. Please don't remove it.

Reply:
Its here: https://github.com/telekom/testerra-selenoid-connector/pull/5/files#diff-2144d51d3560bedcc497ead3188f92cfdce0e658bd62816c6016b2830712c645R70